content/1_docs/2_cookbook/3_templating/0_understanding-oop/recipe.txt

```diff
@@ -0,0 +1,462 @@
+
+The Kirby core is written in PHP, a scripting language that still powers a majority of websites both in the backend and in the frontend.
+
+Like basically all professional PHP applications today, Kirby is based on an OOP (Object Oriented Programming) architecture which has many advantages over the procedural approach you often find in older PHP programs. The idea behind this architecture is to encapsulate data and the methods that work with this data, in order to produce modular, extendable and maintainable code.
+
+<warning>
+No images, lots of text😉. But at least some code to play with. So get some coffee or tea.
+</warning>
+
+## Prerequisites
+
+If you want to code along with this recipe, you need the following:
+
+- A web server (ideally on your (link: docs/cookbook/setup/development-environment text: local machine))
+- A current version of PHP
+- A Kirby installation , either (link: docs/guide/quickstart text: Starterkit or Plainkit)
+- A code/text editor of your choice
+
+## What is OOP?
+
+Object Oriented Programming (OOP) is all about objects, their properties and their methods.
+
+Let's consider a real-world object. Such an object could be a book in your library (or a computer, a smartphone, a piece of furniture…). It has certain properties like an author, a title, an ISBN number, a price tag and a cover, and it was probably published by a publishing house. Its methods would be the actions you could do with the book, e.g. you can read it, you can give it away as a present, put it back in your shelf or throw it away if you don't like it. And while all the books in your library are different, they nevertheless share the same general properties and methods.
+
+OOP uses the same paradigms for data structures in a program. The properties and methods of an object like the book are described in a class, which acts as the model or blueprint for every book you might have in your application, whereas an object is an instance of this class, i.e. a specific embodiment of the class (the specific book in your library).
+
+In a larger software program, you typically have multiple classes whose objects can interact with each other (like a book, a bookshop, the customers that buy the books in your shop, invoices for sold books etc).
+
+### A simple class example
+
+Time for the first coffee. Let's start with a simple example and do some coding to see how we can actually create such a class in PHP and how we can create (instantiate) objects from that class. We define a class called `Book` with some properties that should apply to all objects of this class, and methods to work with the data of these objects.
+
+In your webserver's (glossary: document-root text: document root), create a new folder `book` and inside of this folder a new file called `book.php` with the class definition code from above. Copy the code below into this file.
```

Comment:
I'd name the file `book/Book.php` according to PSR standards so readers can learn it the "correct" way from the start.